src/main/java/org/jabref/gui/WaitForBackgroundtasksFinishedDialog.java

```diff
@@ -0,0 +1,64 @@
+package org.jabref.gui;
+
+import java.util.Optional;
+
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.gui.util.ThemeLoader;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.controlsfx.control.TaskProgressView;
+import org.fxmisc.easybind.EasyBind;
+
+/**
+ * Dialog shown when closing of application needs to wait for some background tasks.
+ */
+public class WaitForBackgroundtasksFinishedDialog {
+
+    private final DialogService dialogService;
+
+    public WaitForBackgroundtasksFinishedDialog(DialogService dialogService) {
+        this.dialogService = dialogService;
+    }
+
+    public boolean showAndWait(StateManager stateManager, ThemeLoader themeLoader, JabRefPreferences preferences) {
+        TaskProgressView taskProgressView = new TaskProgressView();
+        EasyBind.listBind(taskProgressView.getTasks(), stateManager.getBackgroundTasks());
+        taskProgressView.setRetainTasks(false);
+        taskProgressView.setGraphicFactory(BackgroundTask.iconCallback);
+
+        Label message = new Label(Localization.lang("Waiting for background tasks to finish. Quit anyway?"));
+
+        VBox box = new VBox(taskProgressView, message);
+
+        DialogPane contentPane = new DialogPane();
+        contentPane.setContent(box);
+
+        FXDialog alert = new FXDialog(Alert.AlertType.NONE, Localization.lang("Please wait..."));
```

Comment:
https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/gui/DialogService.java#L179 (or the other overload) cannot be used here? I think it would be slightly cleaner to make `WaitForBackgroundtasksFinishedDialog` a "real" dialog (i.e. inherit from `Dialog`) or even convert it to a proper fxml-based dialog similar to most dialogs.

Reply:
I did use the custom dialogue at the beginning, but it gives no access to the actual javafx dialogue, meaning I cannot hide it on an event. The thing with the JabRef dialogues created by the show*AndWait is that they create the actual javafx dialogue in the method, so they can't be accessed later. I just found the createDialog method which would probably be the best to use here. It creates the dialog and takes care of some styling and such, and then returns the actual dialog which can then be shown and waited for. I'll do that!

Reply:
Is this ok? I tried to follow how showProgressDialogAndWait is implemented.
https://github.com/JabRef/jabref/pull/6443/commits/ff9ce0068d6b80e131826a9138aab3b1f981b00e